src/device-state.ts

```diff
@@ -684,24 +684,36 @@ export function reportCurrentState(newState: DeviceReport = {}) {
 	emitAsync('change', undefined);
 }
 
-export async function reboot(force?: boolean, skipLock?: boolean) {
-	await updateLock.abortIfHUPInProgress({ force });
-	await applicationManager.stopAll({ force, skipLock });
-	logger.logSystemMessage('Rebooting', {}, 'Reboot');
-	const $reboot = await dbus.reboot();
-	shuttingDown = true;
-	emitAsync('shutdown', undefined);
-	return await $reboot;
+export interface ShutdownOpts {
+	force?: boolean;
+	reboot?: boolean;
 }
 
-export async function shutdown(force?: boolean, skipLock?: boolean) {
+export async function shutdown({
```

Comment:
I don't like this `shutdown({reboot: true})` signature. A reboot and a shutdown are different things. I understand that the idea is to make the code more DRY, but I think in that case would make more sense to create a function

```
withLocking(fn: () => any,  opts: {force: boolean}) that calls the function passed as argument only if there are no locks on any app, then both reboot and shutdown can call `withLocking` changing just the callback.
```

I understand if you want to leave these changes for another PR though

Reply:
I disagree, they are semantically different, and the operations are different. The interface is not intuitive either, if you are new to the codebase and you want to check how `reboot` is triggered, you won't find it easily.

I'm not going to argue if you want to leave it like this, I'm just giving you my opinion.

Reply:
I'm open to changing it but perhaps can do in another PR.